### PR TITLE
RE batching

### PR DESCRIPTION
**Pull request by msoeken**
Updates notebooks to make use of new RE batching feature. Requires February 2023 QDK.

### CODE REVIEW

File: samples/azure-quantum/resource-estimation/estimation-factoring.ipynb [hunk 1]
```diff
@@ -177,21 +177,23 @@
...
-    "    (\"Gate-based µs, 10⁻³\", {\"qubitParams\": {\"name\": \"qubit_gate_us_e3\"}, \"errorBudget\": 0.333}),\n",
-    "    (\"Gate-based µs, 10⁻⁴\", {\"qubitParams\": {\"name\": \"qubit_gate_us_e4\"}, \"errorBudget\": 0.333}),\n",
-    "    (\"Gate-based ns, 10⁻³\", {\"qubitParams\": {\"name\": \"qubit_gate_ns_e3\"}, \"errorBudget\": 0.333}),\n",
-    "    (\"Gate-based ns, 10⁻⁴\", {\"qubitParams\": {\"name\": \"qubit_gate_ns_e4\"}, \"errorBudget\": 0.333}),\n",
-    "    (\"Majorana ns, 10⁻⁴\", {\"qecScheme\": {\"name\": \"floquet_code\"}, \"qubitParams\": {\"name\": \"qubit_maj_ns_e4\"}, \"errorBudget\": 0.333}),\n",
-    "    (\"Majorana ns, 10⁻⁶\", {\"qecScheme\": {\"name\": \"floquet_code\"}, \"qubitParams\": {\"name\": \"qubit_maj_ns_e6\", \"tGateErrorRate\": 0.01}, \"errorBudget\": 0.333})\n",
+    "labels = [\"Gate-based µs, 10⁻³\", \"Gate-based µs, 10⁻⁴\", \"Gate-based ns, 10⁻³\", \"Gate-based ns, 10⁻⁴\", \"Majorana ns, 10⁻⁴\", \"Majorana ns, 10⁻⁶\"]\n",
+    "\n",
+    "items = [\n",
+    "    {\"qubitParams\": {\"name\": \"qubit_gate_us_e3\"}, \"errorBudget\": 0.333},\n",
```
Comment (katymccl) [inline samples/azure-quantum/resource-estimation/estimation-factoring.ipynb:183]: Is it worth making a constant for error budget?

File: samples/azure-quantum/resource-estimation/integer-factorization-with-cli/jobParamsBatching.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
+    "items": [
```
Comment (katymccl) [inline samples/azure-quantum/resource-estimation/integer-factorization-with-cli/jobParamsBatching.json:2]: Not sure if it is a lot of work to change this word "items", but it doesnt really make sense in the relationship to whats in the array; could be more clear here

**Review by katymccl — APPROVED** [review]
Left some nit comments; would love another person more familiar to review as well but looks good from my end!

File: samples/azure-quantum/resource-estimation/estimation-factoring.ipynb (hunk 1 above)
Comment (msoeken) [reply samples/azure-quantum/resource-estimation/estimation-factoring.ipynb:183]: Items is the terminology to refer to an entity in a batching job, so I prefer to keep that identifier here.

File: samples/azure-quantum/resource-estimation/integer-factorization-with-cli/jobParamsBatching.json (hunk 2 above)
Comment (msoeken) [reply samples/azure-quantum/resource-estimation/integer-factorization-with-cli/jobParamsBatching.json:2]: The keyword `"items"` is to indicate that this is a batching job.

File: samples/azure-quantum/resource-estimation/estimation-factoring.ipynb (hunk 1 above)
Comment (msoeken) [reply samples/azure-quantum/resource-estimation/estimation-factoring.ipynb:183]: I am updating this and the other notebooks in a separate PR.

**Pull request merged**